### DISCUSSION TITLE
Multiple queries for compare system RTK-Query

### DISCUSSION THREAD

**Discussion by andrewdoro** [reduxjs/redux-toolkit#2811] [Q&A]
Hello, I started migrating to RTK and RTK-Query from Redux. Before I was fetching and creating a custom cache for each data response `id` , where on starting the fetch I add the `loadingType:"started"`  in the redux store and when the promise is finished, I change the loadingType to `"finished"`. 

How should I implement this in RTK Query ?

```tsx
 case storeTypes.area.FETCH_DATA: {
      return {
        ...state,
        chains: { ...state.data, [action.payload.id]: { loadingType: 'started' } },
      }
    }
```

``` tsx
 case storeTypes.area.SET_DATA: {
      return {
        ...state,
        data: {
          ...state.data,
          [action.payload.id]: { ...action.payload, loadingType: 'finished' },
        },
      }
    }
```
in the end the object would have looked like this:

``` ts
export type WithLoading<T> =
  | (T & { loadingType: 'finished' })
  | { loadingType: 'started' }
  | { loadingType: 'error' }
...
data: {
   { [key in string]: WithLoading<DataType> }
  }
```

**Comment by phryneas** [comment]
The point of RTK Query is literally to do all of this for you. Just use it as explained in the [tutorial](https://redux.js.org/tutorials/essentials/part-7-rtk-query-basics) and it is doing exactly that internally.

**Reply by andrewdoro** [reply]
yes, I understand what RTQ Query is doing, but in my example I use the system to have all the values in the store for each id and then show a graph. Can I access the rtq query cache to do this, or is it related to this #1171 ?

![image](https://user-images.githubusercontent.com/17547035/197615366-1278e00b-1141-481f-8566-de29b71ac14a.png)
![image](https://user-images.githubusercontent.com/17547035/197615654-d570e849-c4df-45bd-a3cb-c5dd43e9a165.png)
